### ISSUE TITLE
When there are multiple routes for same url, random route is called

### ISSUE THREAD

**Issue by phoenixcz** [cypress-io/cypress#1788]
I am testing a contacts app. In following scenario, I need to load default list of contacts, then perform import and load another list of contacts (including imported contacts) to check that contacts are displayed correctly. 

Similar issue: https://github.com/cypress-io/cypress/issues/85

### Current behavior: 
After the route is overwritten, random route is matched. (sometimes when the test is executed `@getAllContacts1` is matched another time the test is executed `@getAllContacts2` is matched by the same action)

### Desired behavior:
After the route is overwritten, second route `@getAllContacts2` should be matched

### Steps to reproduce:
```js
it('Contacts are selected after import', () => {
  cy.server()
      .route({
        method: 'GET',
        url: contactsUrl,
        response: 'fx: contacts1',
    }).as('getAllContacts1')
      .visit(myAppUrl)
      .wait('@getAllContacts1')
      .get(basePage.importContactButton).click()
      .uploadFile('import.csv', importPage.fileInput)
      //Overwrite default returned contacts with new json including imported contacts
      .route({
        method: 'GET',
        url: contactsUrl,
        response: 'fx: contacts2',
      }).as('getAllContacts2')
      .get(importPage.nextButton).click()
      .wait('@getAllContacts2')
      //Check that contacts are displayed correctly
      .....
});
```

### Versions  

Cypress 2.1.0, Windows 10, Chrome 66

**Comment by jennifer-shehane** [comment]
Could you be more specific by what you mean by 

> random fixture is returned

Could you share what it yielded in the response of the `@getAllContacts2`?

```js
cy.wait('@getAllContacts2').then((xhr) => {
  console.log(xhr.responseBody)
})
```

**Comment by phoenixcz** [comment]
This seems to got resolved by updating to Cypress 3.0.1. Feel free to close this ticket.

**Comment by jennifer-shehane** [comment]
Great to hear @phoenixcz!
